pub/lib/dependabot/pub/update_checker.rb

```diff
@@ -55,6 +77,7 @@ def updated_requirements
       # * If current_version is non-nil and the parsed version is the same it
       #   will be returned.
       # * Otherwise returns nil
+      # TODO pretty sure this should be raisig "AllVersionsIgnored" if version.nil? && @raise_on_ignored"??
```

Comment:
@jonasfj @sigurdm am I reading this code correctly?

I see `AllVersionsIgnored` gets raised within the `latest_version` path:
https://github.com/dependabot/dependabot-core/blob/c5e5b827af80f60482262ffca0bd9a2bb7a4d73d/pub/lib/dependabot/pub/update_checker.rb#L14

But `latest_resolvable_version` never hits that... instead it goes through `version_unless_ignored` which never raises for this condition:
https://github.com/dependabot/dependabot-core/blob/c5e5b827af80f60482262ffca0bd9a2bb7a4d73d/pub/lib/dependabot/pub/update_checker.rb#L51-L71

Is it a bug that `version_unless_ignored` doesn't raise or was that intentional?